great_expectations/core/expectation_validation_result.py

```diff
@@ -123,7 +123,7 @@ def __eq__(self, other):
         #         return NotImplemented
         if not isinstance(other, self.__class__):
             # Delegate comparison to the other instance's __eq__.
-            return NotImplemented
```

Comment:
This raises a `DeprecationWarning` - you can't return `NotImplemented` in a boolean context. I went ahead and implemented what the comment said